File: app/src/main/java/com/fast/todoapp/Model/TaskDao.java
package com.fast.todoapp.Model;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface TaskDao {

    @Insert
    void insert(Task task);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void update(Task task);

    @Delete
    void delete(Task task);

    @Query("DELETE FROM task_table")
    void deleteAllTasks();

    @Query("SELECT * FROM task_table ORDER BY checked ASC")
    LiveData<List<Task>> getTasks();
}

File: app/src/main/java/com/fast/todoapp/data/dao/TaskDao.kt
package com.fast.todoapp.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.fast.todoapp.data.models.TaskEntity

@Dao
interface TaskDao {

    @Insert
    suspend fun insert(entity: TaskEntity)

    @Update
    suspend fun update(entity: TaskEntity)

    @Delete
    suspend fun delete(entity: TaskEntity)

    @Query("DELETE FROM task_table")
    suspend fun deleteAllTasks()

    @Query("SELECT * FROM task_table ORDER BY checked ASC")
    fun getAllTasks(): LiveData<List<TaskEntity>>
}